data_structures/binary_search_tree_OOP_approach.cpp

```diff
@@ -0,0 +1,205 @@
+/**
+ * \file
+ * \brief A simple tree implementation using structured nodes
+ * \Object Oriented approach
+ */
+
+#include<bits/stdc++.h> 
+#include<queue>
+using namespace std;
+
+struct node {
+    int value;
+    node *left;
+    node *right;
+};
+
+class BinarySearchTree{
+    node *root;
+
+public:
+    BinarySearchTree(){
+        root = NULL;
+    }
+
+    void insert(int data){
+        node *newnode = new node;
+        newnode->value = data;
+        newnode->left = NULL;
+        newnode->right = NULL;
+        if(root == NULL){
+            root = newnode;
+            return;
+        }
+        node *current = root;
+        while(true){
+            if(current->value > data){
+                if(current->left == NULL)
+                {
+                    current->left = newnode;
+                    return;
+                }
+                else
+                    current = current->left;
+            }
+            else{
+                if(current->right == NULL)
+                {
+                    current->right = newnode;
+                    return;
+                }
+                else
+                    current = current->right;
```

Comment:
Make it look like C++ code:
```suggestion
                else {
                    current = current->right;
                }
```